### PR TITLE
Sparse hfs migration

### PR DESCRIPTION
**Pull request by multun**

### CODE REVIEW

File: python-lib/prologin/hfs/server.py [hunk 1]
```diff
@@ -337,18 +339,33 @@ def remote_user_handler(self, peer_id):
...
+            while True:
+                block = rfp.read(BLOCK_SIZE)
+                if not block:
+                    break
+                tar.stdin.write(block)
+
+        tar.stdin.close()
+
+        if tar.wait() != 0:
+            raise RuntimeError(f"{' '.join(tar_cmdline)} exited with rc")
```
Comment (audebert) [inline python-lib/prologin/hfs/server.py]: Does it crash the RFS server if this happens?

File: python-lib/prologin/hfs/server.py [hunk 2]
```diff
@@ -337,18 +339,33 @@ def remote_user_handler(self, peer_id):
...
-                    while True:
-                        block = zfp.read(65536)
-                        if not block:
-                            break
-                        fp.write(block)
+        nbd_path = Path(self.nbd_filename())
+
+        tar_cmdline = ['/usr/bin/tar', '--sparse', '-xzf',
+                       '-', nbd_path.name]
+
```
Comment (audebert) [inline python-lib/prologin/hfs/server.py]: nit: remove extra \n

File: python-lib/prologin/hfs/server.py [hunk 3]
```diff
@@ -62,6 +61,8 @@
 import urllib.parse
 import urllib.request
```
Comment (audebert) [inline python-lib/prologin/hfs/server.py]: nit: remove \n

File: python-lib/prologin/hfs/server.py (hunk 1 above)
Comment (multun) [reply python-lib/prologin/hfs/server.py]: nop

**Pull request merged**